services/second-service/src/posts/posts.module.ts

```diff
@@ -3,11 +3,19 @@ import { MongooseModule } from '@nestjs/mongoose';
 import PostsController from './posts.controller';
 import PostsService from './posts.service';
 import { postForFeature } from './post.schema';
+import { PostsDto } from './dto/posts.dto';
+import { CLASS_TRANSFORM_DTO } from '../constants';
 
 @Module({
   imports: [MongooseModule.forFeature([postForFeature])],
   controllers: [PostsController],
-  providers: [PostsService],
+  providers: [
+    PostsService,
+    {
+      provide: CLASS_TRANSFORM_DTO,
+      useValue: PostsDto,
```

Comment:
I would rely on the method return type. We might have different DTOs for different methods. Although global config  is also nice as a default value